quixstreams/topic_manager.py

```diff
@@ -0,0 +1,770 @@
+
+        In general, users should NOT need this; an Application knows when/how to
+        generate changelog topics. To turn off changelogs, init an Application with
+        "use_changelog_topics"=`False`.
+
+        :param consumer_group: name of consumer group (for this app)
+        :param source_topic_name: name of consumed topic (app input topic)
+        :param suffix: name of storage type (default, rolling10s, etc.)
+        :param configs_to_import: what extra_configs should be allowed when importing
+            settings from the source topic.
+
+        :return: `Topic` object (which is also stored on the TopicManager)
+        """
+        # TODO: consider removing configs_to_import as changelog settings management
+        # around retention, quix compact settings, etc. matures.
+        ...
+
+    @abstractmethod
+    def create_all_topics(self):
+        """
+        A convenience method to create all Topic objects stored on this TopicManager.
+        """
+        ...
+
+    @abstractmethod
+    def validate_all_topics(
+        self,
+        validation_level: Literal["exists", "required", "all"] = "exists",
+    ):
+        """
+        A convenience method for validating all `Topic`s stored on this TopicManager.
+
+        See `TopicManager.validate_topics()` for more details.
+
+        :param validation_level: The degree of topic validation; Default - "exists"
+            "exists" - Confirm expected topics exist.
+            "required" - Confirm topics match your provided `Topic`
+                partition + replication factor
+            "all" - Confirm topic settings are EXACT.
+        """
+        ...
+
+
+class TopicManagerBase(TopicManagerType, Protocol):
+    """
+    Defines all `TopicManagerType`s required `abstractmethod`s while also
+    adding some new ones; these new `abstractmethod`s conveniently highlight the
+    differences between `TopicManager` and `QuixTopicManager`.
+    """
+
+    @abstractmethod
+    def _apply_topic_prefix(self, name: str) -> str:
+        """
+        Apply a prefix to the given name
+
+        :param name: topic name
+
+        :return: name with added prefix (no change in this case)
+        """
+        ...
+
+    @abstractmethod
+    def _format_changelog_name(
+        self, consumer_group: str, source_topic_name: str, suffix: str
+    ):
+        """
+        Generate the name of the changelog topic based on the following parameters.
+
+        This naming scheme guarantees uniqueness across all independent `Application`s.
+
+        :param consumer_group: name of consumer group (for this app)
+        :param source_topic_name: name of consumed topic (app input topic)
+        :param suffix: name of storage type (default, rolling10s, etc.)
+
+        :return: formatted topic name
+        """
+        ...
+
+    @abstractmethod
+    def _create_topics(self, topics: TopicList):
+        """
+        Method that actually creates the topics in Kafka via an `Admin` instance.
+
```

Comment:
if we change the method name in `Admin`, we should change it here as well to `ensure_topic_created` to similar